tutorials/3d/procedural_geometry/surfacetool.rst

```diff
@@ -76,7 +86,11 @@ to remove duplicate vertices.
     st.add_index(3)
     st.add_index(2)
 
-    # Alternatively:
+    # If don't want to use add_index() but just index(), you need to add six
+    # vertices via add_vertex() (three for each triangle corner).
+    # index() will then take care of generating the correct array index.
+    # it will also remove duplicate vertices where all attributes,
+    # including the position are the same.
```

Comment:
Instead of this long paragraph, we should modify the example to be more intuitive. 

One way of doing that is to fill out the example a little more. 

I recommend adding the following just above ``# Creates a quad from four corner vertices.``:

```
# Suppose we have a quad defined by 6 vertices as follows
st.add_vertex(Vector3(-1, 1, 0))
st.add_vertex(Vector3(1, 1, 0))
st.add_vertex(Vector3(-1, -1, 0))

st.add_vertex(Vector3(1, 1, 0))
st.add_vertex(Vector3(1, -1, 0))
st.add_vertex(Vector3(-1, -1, 0))

# We can make the quad more efficient by using an index array and only utilizing 4 vertices

```

Then we can replace the long comment with:

```suggestion
    # Alternatively we can use ``st.index()`` which will create the quad for us and remove the duplicate vertices
```